File: backend/app/api/threat_intel.py
from fastapi import APIRouter, Depends

from app.api.dependencies import get_current_user
from app.schemas.threat_intel import (
    ThreatLookupRequest,
    ThreatLookupResponse,
)
from app.services.threat_intel_service import lookup_threat_indicator

router = APIRouter(
    prefix="/threat-intel",
    tags=["Threat Intelligence"],
)

@router.post(
    "/lookup",
    response_model=ThreatLookupResponse,
)
def threat_lookup(
    payload: ThreatLookupRequest,
    current_user=Depends(get_current_user),
):
    return lookup_threat_indicator(
        indicator=payload.indicator,
        indicator_type=payload.type,
    )

File: frontend/src/services/threatIntel.ts
import api from "./api";

import type {
  IndicatorType,
  ThreatLookupResponse,
} from "../types/threatIntel";

export async function lookupThreatIndicator(
  indicator: string,
  type: IndicatorType,
): Promise<ThreatLookupResponse> {
  const response = await api.post(
    "/threat-intel/lookup",
    {
      indicator,
      type,
    },
  );

  return response.data;
}
